### PR TITLE
[multi-cluster] Resolve clusters available to the adjacent istio control plane

### PR DESCRIPTION
**Pull request by israel-hdez**
Add an endpoint to retrieve the clusters that are visible to the adjacent control plane.

Given that Kiali is doing it's own resolution, there is no guarantee that the resolved list is what Istio is seeing. Because of misconfigurations, Kiali may recognize a different list to what Istio has registered internally. But this is the best effort we can do, for now.

Related #3409
Front-end: kiali/kiali-ui#2000

### CODE REVIEW

**Comment by israel-hdez** [comment]
@jshaughn I think this is ready for a review.
Tests are pending.

File: kubernetes/secret.go
Comment (xeviknal) [inline kubernetes/secret.go:54]: For me, it seems more natural to have this methods into the `client.go` file because there are all the methods dealing with pure k8s objects. This secret files sounds a bit more a helpers around dealing with secrets.

File: business/layer.go
Comment (xeviknal) [inline business/layer.go:129]: Yep, I think it is safe now to re-sort the services. We had problems in the past with that because of some services including other services instead of the whole temporary layer. So :+1:

**Review by xeviknal — COMMENTED** [review]
Looks pretty straightforward :+1: There is one question I'd like to hear from @israel-hdez though.

File: business/clustering.go
Comment (jshaughn) [inline business/clustering.go:10]: I wonder if we maybe we should just call this MeshService as a way to query anything we find relevant to a Mesh.

File: business/clustering.go
Comment (jshaughn) [inline business/clustering.go:17]: Could we just call this Cluster?

File: business/clustering.go
Comment (jshaughn) [inline business/clustering.go:22]: Maybe we could be explicit with this and a few other characterizations of the cluster, like HasKiali, HasControlPlane, HasDataPlane etc?

File: business/clustering.go
Comment (jshaughn) [inline business/clustering.go:54]: Again, we could just be explicit, and instead of inferring the Kiali point of view in the API, just resolveKialiControlPlaneCluster().

**Review by jshaughn — COMMENTED** [review]
Functionally it looks OK to me.  Organizationally, Xavi's comment looks good from a consistency standpoint.  For me I think maybe the terminology could be a cleaner but it is not a blocker.

One thing I wonder is whether we should initially work in a branch as we explore this feature, get feedback, etc.

File: business/clustering.go
Comment (israel-hdez) [reply business/clustering.go:10]: Yes, I think your suggested renaming is OK, as this may include some things not related to clustering. I'll do it.

File: kubernetes/secret.go
Comment (israel-hdez) [reply kubernetes/secret.go:54]: I think I'm not sure to understand the structure that good :) because `client.go` for me, looks more like interface definitions and a few helpers related to creating the client. So, I this code would be like an "outlier" if moved to `client.go` :/

Perhaps, it's something else?

File: business/clustering.go
Comment (israel-hdez) [reply business/clustering.go:22]: Hmm... Naming is hard here...

So, in my env I have two clusters. Both can be marked as HasKiali, HasControlPlane and HasDataPlane, yet only one should be marked with "IsHomeCluster" and the one that gets this flag depends completely on which Kiali instance you are accessing...

So, all three variables HasKiali, HasControlPlane and HasDataPlane are good ideas to have, but this PR is still not finding this data, so I'll not add them yet (but I'll try to add in follow-ups). 

Focusing on "IsHomeCluster"... well... It could be named like `IsHostingThisKialiInstance`???? .... I'm now sure, I don't have ideas of a good naming here. May be the comment is what I should fix?

File: business/clustering.go
Comment (israel-hdez) [reply business/clustering.go:54]: OK, I'll rename this.

File: business/clustering.go
Comment (jshaughn) [reply business/clustering.go:22]: I see your point.  A couple of ideas, isHomeKiali, isCurrentKiali, isActiveKiali, isSessionKiali.

**Comment by israel-hdez** [comment]
@jshaughn Alike in the front-end, I did the renamings.

Same here... once I get the approvals, I'll close this PR and open a new one targeting a new branch that we can use to work on the MC epic (and also I'll fix conflicts there).

**Review by jshaughn — APPROVED** [review]
@israel-hdez OK, this looks good.  But we need to merge it into a feature branch, not into master.  We can work in a feature branch until we're satisfied with approach, details. etc.

**Comment by israel-hdez** [comment]
@jshaughn OK, I changed the target branch.
Also, I added a new commit adding tests, just in case you want to review the tests.

**Comment by israel-hdez** [comment]
As this is targeting a feature branch, I'm merging. But I still would like to get clarification of @xeviknal comment to move code in a follow-up PR, if needed.

**Pull request merged**
